src/rust.rs

```diff
@@ -379,19 +376,20 @@ impl Runtime {
                     -> Result<(),()> {
         let filename_cstr = ffi::CString::new(filename.as_bytes()).unwrap();
         debug!("Evaluating script from {} with content {}", filename, script);
-        // SpiderMonkey does not approve of null pointers.
-        let (ptr, len) = if script.len() == 0 {
-            static empty: &'static [u8] = &[];
-            (empty.as_ptr(), 0)
-        } else {
-            (script.as_bytes().as_ptr(), script.len() as c_uint)
-        };
-        assert!(!ptr.is_null());
```

Comment:
It got over its dislike of `null`?

Reply:
Empty slices have not been the `null` pointer for years.

Reply:
And strings etc I mean, an `&str` with a `null` pointer is UB and has been for many years.